core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -792,23 +792,17 @@ class ZkUtils(val zkClient: ZkClient,
   /**
     * This API produces a sequence number by creating / updating given path in zookeeper
     * It uses the stat returned by the zookeeper and return the version. Every time
-    * client updates the path stat.version gets incremented
+    * client updates the path stat.version gets incremented. Starting value of sequence number is 1.
     */
   def getSequenceId(path: String, acls: java.util.List[ACL] = DefaultAcls): Int = {
     try {
       val stat = zkClient.writeDataReturnStat(path, "", -1)
       stat.getVersion
     } catch {
       case e: ZkNoNodeException => {
-        createParentPath(BrokerSequenceIdPath, acls)
-        try {
-          zkClient.createPersistent(BrokerSequenceIdPath, "", acls)
-          0
-        } catch {
-          case e: ZkNodeExistsException =>
-            val stat = zkClient.writeDataReturnStat(BrokerSequenceIdPath, "", -1)
-            stat.getVersion
-        }
+        makeSurePersistentPathExists(path)
+        val stat = zkClient.writeDataReturnStat(path, "", -1)
```

Comment:
Can we avoid the two occurrences of `zkClient.writeDataReturnStat(path, "", -1)`? One way suggested by @ijuma offline is to declare an inner method:

```
def writeToZk: Int = zkClient.writeDataReturnStat(path, "", -1).getVersion
```